src/Adapter/State/AbstractStateHandler.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\State;
+
+use PrestaShop\PrestaShop\Core\Domain\State\Exception\DeleteStateException;
+use PrestaShop\PrestaShop\Core\Domain\State\Exception\StateException;
+use PrestaShop\PrestaShop\Core\Domain\State\Exception\StateNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\State\ValueObject\StateId;
+use PrestaShopException;
+use State;
+
+/**
+ * Abstract state handler provides common methods for state handlers
+ */
+abstract class AbstractStateHandler
+{
+    /**
+     * @param StateId $stateId
+     *
+     * @return State
+     *
+     * @throws StateNotFoundException
+     */
+    protected function getState(StateId $stateId): State
+    {
+        $stateIdValue = $stateId->getValue();
+
+        try {
+            $state = new State($stateIdValue);
+        } catch (PrestaShopException $e) {
+            throw new StateNotFoundException(sprintf('State with id "%s" was not found.', $stateId->getValue()));
+        }
+
+        if ($state->id !== $stateId->getValue()) {
+            throw new StateNotFoundException(sprintf('State with id "%s" was not found.', $stateId->getValue()));
+        }
+
+        return $state;
+    }
+
+    /**
+     * Deletes legacy State
+     *
+     * @param State $state
+     *
+     * @return bool
+     *
+     * @throws DeleteStateException
+     */
+    protected function deleteState(State $state): bool
+    {
+        try {
+            return $state->delete();
+        } catch (PrestaShopException $e) {
+            throw new DeleteStateException(sprintf('An error occurred when deleting State object with id "%s".', $state->id));
+        }
+    }
+
+    /**
+     * Toggles legacy state status
+     *
+     * @param State $state
+     * @param bool $newStatus
+     *
+     * @return bool
+     *
+     * @throws StateException
+     */
+    protected function toggleStateStatus(State $state, bool $newStatus): bool
```

Comment:
It seems there is a `toggleStatus()` on ObjectModel, so maybe this method is useless ?

Reply:
Yes, also to could write Behats for these handlers when https://github.com/PrestaShop/PrestaShop/pull/15071/commits/20ea99521790eff52072ea157438eac4b757bd50 is merged. Started location suite there.

Reply:
As we talked #17407 this approach is better because objectmodel::toggleStatus does set the status passed.